astropy/io/ascii/ecsv.py

```diff
@@ -174,7 +202,8 @@ def get_cols(self, lines):
 
             col.dtype = header_cols[col.name]['datatype']
             if col.dtype not in ECSV_DATATYPES:
```

Comment:
```suggestion
            # Require col dtype to be a valid ECSV datatype. However, older versions
            # of astropy writing ECSV version 0.9 and earlier had inadvertently allowed
            # numpy datatypes like datetime64 or object which are not in the ECSV standard. 
            # For back-compatibility with those existing older files, allow reading with no error.
            if col.dtype not in ECSV_DATATYPES and self.ecsv_version > (0, 9, 0):
```

Reply:
I'm adding this in, but out of curiosity, were there any non-astropy packages that created ECSV versions between 0.9.0 and 1.0.0, which would be missed by this?